src/Application/Videos/Commands/UpdateVideoCommand.cs

```diff
@@ -4,29 +4,33 @@
 
 namespace Videoteka.Application.Videos.Commands;
 
-    public record UpdateVideoCommand(Guid Id, string Name, string? Description) : IRequest;
+    public record UpdateVideoCommand(Guid Id, string Name, string? Description) : IRequest, IAuthorizedRequest
+    {
+        public async Task<bool> Authorize(IApplicationDbContext context, IAuthService authService)
+        {
+            return await authService.ValidateCookieAndGetUsername() != null;
```

Comment:
var video = await context.Videos.Include(x = x.Uploader).FirstAsync(x => x.Id == Id);
var username = await authService.ValidateCookieAndGetUsername();

return username != null && video.Uploader.Username == username;

Reply:
this will make sure the user is not only logged in, but is the uploader of the video